pkg/controller/upgradeconfig/cluster_upgrader.go

```diff
@@ -418,23 +424,24 @@ func RemoveMaintWindow(c client.Client, m maintenance.Maintenance, upgradeConfig
 // This perform cluster health check after upgrade
 func PostClusterHealthCheck(c client.Client, m maintenance.Maintenance, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
 
-	return performClusterHealthCheck(c)
+	return performClusterHealthCheck(c, upgradeConfig)
 }
 
 // Check whether nodes are upgraded or not
-func NodesUpgraded(c client.Client, nodeType string, reqLogger logr.Logger) (bool, error) {
+func NodesUpgraded(c client.Client, nodeType string, upgradeConfig *upgradev1alpha1.UpgradeConfig, reqLogger logr.Logger) (bool, error) {
 	configPool := &machineconfigapi.MachineConfigPool{}
 	err := c.Get(context.TODO(), types.NamespacedName{Name: nodeType}, configPool)
 	if err != nil {
 		return false, nil
 	}
-	//TODO send timeout alert if wait timeout
 	if configPool.Status.MachineCount != configPool.Status.UpdatedMachineCount {
 		errMsg := fmt.Sprintf("not all %s are upgraded, upgraded: %v, total: %v", nodeType, configPool.Status.UpdatedMachineCount, configPool.Status.MachineCount)
 		reqLogger.Info(errMsg)
 		return false, nil
 	}
 
+	// send node upgrade complete metrics
+	metrics.UpdateMetricNodeUpgradeEndTime(time.Now(), upgradeConfig.Name)
```

Comment:
See my comment elsewhere in the review.
```suggestion
	metrics.UpdateMetricNodeUpgradeEndTime(time.Now().UTC(), upgradeConfig.Name)
```